src/icq/poller.rs

```diff
@@ -139,23 +143,49 @@ pub async fn process_event_hist_dlg_state(
     let chat_sn = event_data.sn.clone();
     let chat_friendly = find_author_friendly(&chat_sn, &event_data.persons).to_string();
 
+    let chat_info = ChatInfo {
+        sn: chat_sn.clone(),
+        stamp: None,
+        title: chat_friendly,
+        group: None,
+    };
+
     tx.handle_proxy(&account_info.handle)
         .exec(move |plugin, protocol_data| {
             let connection = &mut protocol_data.connection;
-            let chat_info = ChatInfo {
-                sn: chat_sn,
-                stamp: None,
-                title: chat_friendly,
-                group: None,
-            };
             plugin.chat_joined(connection, &chat_info);
         })
         .await;
 
     // Create Chat Entries
     for message in &event_data.messages {
-        let author_sn = &message.chat.sender;
-        let _author_friendly = find_author_friendly(author_sn, &event_data.persons);
-        //chat.add_message(message)
+        // For group conversation:
+        // - event_data.sn is the Group's sn
+        // - message.chat.sender is the author's sn
+        //
+        // For DMs:
+        // - message.chat is not there
+        // - event_data.sn is the author's sn
+        let author_sn = {
+            match &message.chat {
+                Some(chat) => chat.sender.clone(),
+                None => event_data.sn.clone(),
+            }
+        };
+        let author_friendly = find_author_friendly(&author_sn, &event_data.persons).to_string();
+
+        let chat_input = ChatInput {
+            chat_sn: chat_sn.clone(),
+            author_sn,
+            author_friendly,
+            text: message.text.clone(),
+            time: message.time,
+        };
+
+        tx.handle_proxy(&account_info.handle)
+            .exec(move |_plugin, protocol_data| {
```

Comment:
nit, if you only use connection then `connection_proxy` can be used.

Reply:
ah yes. Done